### PR TITLE
rework basic_info map key to include date

### PR DESCRIPTION
**Pull request by danrubel**

### CODE REVIEW

File: sintr_worker/lib/basic_info.dart [hunk 1]
```diff
@@ -87,13 +86,31 @@ String logKey, Stream<String> stream) async {
...
       return result;
     }
   }
-  result[basic_info_key] = [unknown_value];
+  result[key_prefix] = [unknown_value];
   return result;
 }
+
+/// Return a key of the form <prefix><YYYY>-<MM>-<DD>
+String buildKey(String prefix, int milliSinceEpoch) {
```
Comment (lukechurch) [inline sintr_worker/lib/basic_info.dart]: Add a todo to consider something like this: http://stackoverflow.com/questions/16126579/how-do-i-format-a-date-with-dart

when we have the inftastructure in place to do pub-deployments to the worker nodes?

File: sintr_worker/lib/basic_info.dart [hunk 2]
```diff
@@ -87,13 +86,31 @@ String logKey, Stream<String> stream) async {
...
+  var time = new DateTime.fromMillisecondsSinceEpoch(milliSinceEpoch);
+  var year = time.year.toString();
+  var month = time.month.toString();
+  var day = time.day.toString();
+  if (month.length < 2) month = '0$month';
+  if (day.length < 2) day = '0$day';
+  return '$prefix$year-$month-$day';
+}
```
Comment (lukechurch) [inline sintr_worker/lib/basic_info.dart]: Any chance of a terminal line break? I use cat a lot during development, it does really help

File: sintr_worker/lib/basic_info.dart [hunk 3]
```diff
@@ -87,13 +86,31 @@ String logKey, Stream<String> stream) async {
...
       }
     }
-    if (line.startsWith(data_prefix)) {
-      // Extract the info and add a result of the format
-      // <millisSinceEpoch>:Ver:<uuid>:<clientId>:<clientVersion>:<serverVersion>:<sdkVersion>
-      result[basic_info_key] = [line.substring(data_prefix.length)];
+    if (line.startsWith('Data :: ~')) {
+      int index = line.indexOf(':Ver:');
+      if (index != -1) {
+        int milliSinceEpoch = int.parse(line.substring(9, index));
```
Comment (lukechurch) [inline sintr_worker/lib/basic_info.dart]: This means that in that on the 20th of November, 2286, in the afternoon, the code will stop working.

:(

[/joke]

**Comment by lukechurch** [comment]
lgtm. This is usually a good pattern to follow

File: sintr_worker/lib/basic_info.dart (hunk 1 above)
Comment (danrubel) [inline sintr_worker/lib/basic_info.dart]: Done.

File: sintr_worker/lib/basic_info.dart (hunk 2 above)
Comment (danrubel) [inline sintr_worker/lib/basic_info.dart]: I try to always do that, but missed here. Done.

**Pull request merged**